sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/README.md

```diff
@@ -1,25 +1,305 @@
+```
+
+#### Using Cosmos Tables
+
+To create a new Cosmos Tables , you can use the [Azure Portal][storage_account_create_portal],
+[Azure PowerShell][storage_account_create_ps], or the [Azure CLI][storage_account_create_cli].
+
+### Authenticate the client
+
+Connection represents a set of information required to connect to a table service. It can contain a connection string, an endpoint, token credential or a shared key.
+
+The `Connection` property of `TableAttribute` defines which connection is used for the Table Service access. For example, `[Tables(Connection="MyTableService")]` is going to use `MyTableService` connection.
+
+The connection information can be set in [`local.settings.json`][local_settings_json] or [application settings in Azure portal][appsettings_portal].
+
+When adding a setting to *local.settings.json* place it under the `Values` property:
+
+```json
+{
+  "IsEncrypted": false,
+  "Values": {
+    "MyTableService": "..."
+  }
+}
+```
+
+When adding a setting to application settings in Azure portal use the provided name directly:
+
+`MyTableService = ...`
+
+Tables extension uses the `AzureWebJobsStorage` connection name by default.
+
+#### Connection string
+
+To use connection strings authentication assign connection string value directly to the connection setting.
+
+`<ConnectionName>` = `DefaultEndpointsProtocol=https;AccountName=...;AccountKey=...;EndpointSuffix=core.windows.net`
+
+#### Using endpoint and token credential
+
+**NOTE:** token credential authentication is supported only for storage tables.
+
+`<ConnectionName>__endpoint` = `https://...table.core.windows.net`
+
+If not credential information is provided the [`DefaultAzureCredential`][identity_dac] is used.
+
+When using user-assigned manageed identity the `clientId` and `credential` settings need to be provided:
+
+`<ConnectionName>__credential` = `managedidentity`
+
+`<ConnectionName>__clientId` = `<user-assigned client id>`
+
+#### Using shared key credential
+
+When using [shared key authentication](https://docs.microsoft.com/rest/api/storageservices/authorize-with-shared-key) the `endpoint`, `accountKey` and `accountName` need to be provided.
+
+`<ConnectionName>__endpoint` = `https://...table.core.windows.net`
+
+`<ConnectionName>__credential__accountName` = `<account name>`
+
+`<ConnectionName>__credential__accountKey` = `<account key>`
 
 ## Key concepts
 
-### Thread safety
+The input binding allows you to read table as input to an Azure Function. The output binding allows you to modify and delete table rows in an Azure Function.
+
+Please follow the [input binding tutorial](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table-input?tabs=csharp) and [output binding tutorial](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table-output?tabs=csharp) to learn about using this extension for accessing table service.
+
+## Examples
+
+### Binding to a single entity
```

Comment:
Does this get triggered when that entity is added/modified/deleted? If so, let's add a sentence here to describe that.

Reply:
Nope, it's not a trigger, just a binding. It can only load and save entities.

Reply:
Can we explain a bit in the sample what it would be used for?